src/ui/components/filters/filteroptionscomponent.js

```diff
@@ -337,25 +337,35 @@ export default class FilterOptionsComponent extends Component {
 
     // searchable option list
     if (this.config.searchable) {
+      const clearSearchEl = DOM.query(this._container, '.js-yxt-FilterOptions-clearSearch');
+      const searchInputEl = DOM.query(this._container, '.js-yxt-FilterOptions-filter');
       const filterOptionEls = DOM.queryAll(this._container, '.js-yxt-FilterOptions-option');
       const filterContainerEl = DOM.query(this._container, '.js-yxt-FilterOptions-container');
+
+      // On clearSearchEl click, clear search
+      if (clearSearchEl && searchInputEl) {
+        DOM.on(clearSearchEl, 'click', event => {
+          searchInputEl.value = '';
+          searchInputEl.focus();
+          this._clearFilterStyling(filterOptionEls, filterContainerEl, clearSearchEl);
+        });
+      }
+
       DOM.on(
-        DOM.query(this._container, '.js-yxt-FilterOptions-filter'),
+        searchInputEl,
         'keyup',
```

Comment:
Minor thing, but could we listen for an 'input' event here instead? That could allow us to consolidate a bit. In the handler for clicking the clearSearchEl, we wouldn't need to call this._clearFilterStyling (I think). We'd rely on this block for that.

Reply:
Ah, yes, that is much better. Updating!

Reply:
I was testing this, requires firing an "input" event manually but is much cleaner. Was able to remove the majority of the JS changes previously in the PR.